numpy/core/src/umath/ufunc_object.c

```diff
@@ -5893,13 +5893,45 @@ new_array_op(PyArrayObject *op_array, char *data)
     return (PyArrayObject *)r;
 }
 
+/*
+ * If operands are 1D we can use the indexed loop to do the work
+ * Returns 0 if successful, -1 otherwise
+ */
+static int
+try_trivial_at_loop(PyArrayMethodObject *ufuncimpl, NPY_ARRAYMETHOD_FLAGS flags,
+                    PyArrayMapIterObject *iter,
+                    PyArrayObject *op1_array, PyArrayObject *op2_array,
+                    PyArrayMethod_Context *context)
+{
+    if (PyArray_NDIM(op1_array) != 1) {
+        return -1;
+    }
+    /* check that nop == 3 */
+    if (ufuncimpl->nin >1 && PyArray_NDIM(op2_array) != 1) {
+        return -1;
+    }
+    if (iter->numiter > 1) {
+        return -1;
+    }
+    char *args[3];
+    npy_intp dimensions = iter->size;
+    npy_intp steps[3];
+    args[0] = (char *) iter->baseoffset;
+    args[1] = (char *) iter->outer_ptrs[0];
+    args[2] = (char *)PyArray_DATA(op2_array);
+    steps[0] = iter->fancy_strides[0];
+    steps[1] = iter->outer_strides[0];
+    steps[2] = PyArray_STRIDE(op2_array, 0);
+    ufuncimpl->contiguous_indexed_loop(context, args, &dimensions, steps, NULL);
```

Comment:
We have return value, and it should be negative for failure.  A three-way return (-1 for failure, and 0/1 for whether the trivial path can be taken) would make sense maybe?

Reply:
resolved by a refactor, which also added missing parts to handle warnings and errors